ada_feeding/ada_feeding/behaviors/transfer/planning_scene.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+This module defines behaviors that manipulate the planning scene during bite
+transfer.
+"""
+
+# Standard imports
+from typing import Optional, Union
+
+# Third-party imports
+from geometry_msgs.msg import (
+    Point,
+    PointStamped,
+    Pose,
+    PoseStamped,
+    Quaternion,
+    QuaternionStamped,
+)
+from overrides import override
+import py_trees
+from std_msgs.msg import Header
+
+# Local imports
+from ada_feeding.behaviors import BlackboardBehavior
+from ada_feeding.helpers import BlackboardKey
+
+
+class ComputeWheelchairCollisionTransform(BlackboardBehavior):
+    """
+    A behavior that takes in the detected head pose, the original head pose,
+    and the original wheelchair_collision pose, and computes the new
+    wheelchair_collision pose and scale.
+
+    NOTE: Although this class is in theory rich enough to compute arbitrary
+    updates to the wheelchair collision object, e.g., moving its x and y to be
+    centered on the face, currently we don't translate it at all and only scale
+    z in (0.0, inf). This is because incorporating translations may bring the
+    wheelchair collision object into collision with the robot base, which is not desirable.
+    """
+
+    # pylint: disable=arguments-differ
+    # We *intentionally* violate Liskov Substitution Princple
+    # in that blackboard config (inputs + outputs) are not
+    # meant to be called in a generic setting.
+
+    def blackboard_inputs(
+        self,
+        detected_head_pose: Union[BlackboardKey, PoseStamped],
+        # NOTE: The below parameters must match the ones in `ada_planning_scene.yaml`
```

Comment:
In theory, shouldn't you be able to pull the parameters from the `ada_planning_scene` node? You can use this as a default, but I believe the tree logic should do a ROS param to blackboard write and then fill that in here.

Either fix or flag as an issue.

Reply:
Good point re. Getting parameters via service call.

Although what I think may be a better architecture is to do everything in ada_planning_scene.py. Currently, that code initialise the planning scene and then dies. But instead, we can have it subscribe to the results of face detection (which is only published when it is toggled on) and update the head and wheelchair collision objects. Thus, all planning scene logic is handled in ada_planning_scene.py.

What do you think? I believe there is already a TODO for this change.

Reply:
I agree that keeping all the planning scene logic in ada_planning_scene.py makes sense.

Reply:
Cool, I also added an issue for that, and will merge this in